source/core/slang-shared-library.h

```diff
@@ -0,0 +1,122 @@
+#ifndef SLANG_SHARED_LIBRARY_H_INCLUDED
+#define SLANG_SHARED_LIBRARY_H_INCLUDED
+
+#include "../../slang.h"
+#include "../../slang-com-helper.h"
+#include "../../slang-com-ptr.h"
+
+#include "../core/platform.h"
+#include "../core/common.h"
+#include "../core/dictionary.h"
+
+namespace Slang
+{
+
+/* NOTE! Do not change this enum without making the appropriate changes to DefaultSharedLibraryLoader::s_libraryNames */
+enum class SharedLibraryType
+{
+    Unknown,            ///< Unknown compiler
+    Dxc,                ///< Dxc compiler
```

Comment:
As you know, our naming conventions are currently all over the map. I am currently slightly leaning in favor of keeping acronyms/initialisms in all caps, to match how they are conventionally written in prose. I guess in a case like this it potentially creates confusion about whether `DXC` would be a macro or not, but given `SharedLibraryType::DXC` it seems reasonably obvious.

Not requesting any change to this PR, of course. There is no point enforcing conventions until we've documented them (which I'm guessing we need to do soon).

Reply:
I used to think the caps was the right way. But another place I worked they had upper camel for acronyms - and there I came to think that it was significantly better from readability point of view. Apparently Microsoft has performed a study showing this result. The readability issue is especially so if you have acronym followed by acronym, but it's still bad in general, because if you are using upper camel, the first letter of the next word, blends into the acronym. 
Microsoft does recommend capitalizing two letter acronyms tho - which seems a bad exception.
https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-1.1/141e06ef(v=vs.71)

More code in slang is probably caps, so this is arguably a slip up on my front. So if all caps is the decided way, it can of course be changed.